### PR TITLE
Fix build failure with new method to get pid

### PR DESCRIPTION
**Pull request by lankavitharana**

### CODE REVIEW

**Review by yasassri — APPROVED** [review]

File: modules/tests/test-integration/src/test/java/org/ballerinalang/test/context/ServerInstance.java [hunk 1]
```diff
@@ -441,33 +443,28 @@ private String getServerPID() throws BallerinaTestException {
...
-            FileReader fileReader = null;
-
+            //reading the process id from netstat
+            Process tmp;
             try {
-                //reading the pid form  carbon.pid file in server home dir
-                fileReader = new FileReader(serverHome + File.separator + Constant.SERVER_PID_FILE_NAME);
-                bufferedReader = new BufferedReader(fileReader);
-                pid = bufferedReader.readLine();
+                tmp = Runtime.getRuntime().exec("lsof -Pi tcp:" + httpServerPort);
```
Comment (yasassri) [inline modules/tests/test-integration/src/test/java/org/ballerinalang/test/context/ServerInstance.java]: lsof -ti tcp:9090 will give the PID, Is there any reason for not using this?

File: modules/tests/test-integration/src/test/java/org/ballerinalang/test/context/ServerInstance.java [hunk 2]
```diff
@@ -441,33 +443,28 @@ private String getServerPID() throws BallerinaTestException {
...
-                    }
+                throw new BallerinaTestException("Error retrieving netstat data", e);
+            }
+
+            String outPut = readProcessInputStream(tmp.getInputStream());
+            String[] lines = outPut.split(System.lineSeparator());
+            for (String line : lines) {
+                String[] column = line.trim().split("\\s+");
+                if (column.length < 10) {
+                    continue;
```
Comment (chanakaudaya) [inline modules/tests/test-integration/src/test/java/org/ballerinalang/test/context/ServerInstance.java]: @lankavitharana Check the command mentioned by @yasassri . It directly gives the PID. you don't need to do the string manipulations which you have done here.

**Review by chanakaudaya — CHANGES_REQUESTED** [review]

File: modules/tests/test-integration/src/test/java/org/ballerinalang/test/context/ServerInstance.java (hunk 1 above)
Comment (lankavitharana) [inline modules/tests/test-integration/src/test/java/org/ballerinalang/test/context/ServerInstance.java]: no special reason, but when trying to list processes using the given port, it shows both listening processes and also connected processes, hence need to filter only the listening process.

File: modules/tests/test-integration/src/test/java/org/ballerinalang/test/context/ServerInstance.java (hunk 2 above)
Comment (lankavitharana) [inline modules/tests/test-integration/src/test/java/org/ballerinalang/test/context/ServerInstance.java]: checked that, I think we can't even use that, the reason is, it too gives both listening and connected processes PIDs and we cannot distinguish which is which( there may be other best approaches out there to do this directly, but didn't came across anything yet)

Comment (lankavitharana) [inline modules/tests/test-integration/src/test/java/org/ballerinalang/test/context/ServerInstance.java]: And I think we should do some refactoring to current integration tests, this PR is a quick solution to the current tools distribution build failure

File: modules/tests/test-integration/src/test/java/org/ballerinalang/test/context/ServerInstance.java (hunk 1 above)
Comment (yasassri) [inline modules/tests/test-integration/src/test/java/org/ballerinalang/test/context/ServerInstance.java]: @lankavitharana Shall we try the below command,
```
String[] cmd = { "/bin/sh", "-c",
                        "lsof -Pi tcp:" + httpServerPort + " | grep LISTEN | awk \'{print $2}\'" };
tmp = Runtime.getRuntime().exec(cmd);
```
and then to get the PID,
```
 String outPut = readProcessInputStream(tmp.getInputStream());
            pid = outPut.split("\n")[0];
```

Comment (lankavitharana) [inline modules/tests/test-integration/src/test/java/org/ballerinalang/test/context/ServerInstance.java]: Added that thanks @yasassri  for the point on how to use grep within exec command

**Review by chanakaudaya — APPROVED** [review]

**Pull request merged**